crates/shared/src/price_estimation/competition.rs

```diff
@@ -62,18 +63,28 @@ impl RacingCompetitionPriceEstimator {
             successful_results_for_early_return,
         }
     }
-}
 
-impl PriceEstimating for RacingCompetitionPriceEstimator {
-    fn estimate(&self, query: Arc<Query>) -> futures::future::BoxFuture<'_, PriceEstimateResult> {
+    fn estimate_generic<
```

Comment:
Fancy... The `Debug` constraint feels unnecessary (but is nice for debugging and logging purposes) so probably worth keeping around.